base/runtime/virtual_memory_darwin.odin

```diff
@@ -0,0 +1,94 @@
+#+private
+package runtime
+
+import "base:intrinsics"
+
+VIRTUAL_MEMORY_SUPPORTED :: true
+
+foreign import lib "system:System.framework"
+
+foreign lib {
+	mach_task_self :: proc() -> u32 ---
+	mach_vm_deallocate :: proc(target: u32, address: u64, size: u64) -> i32 ---
+	mach_vm_map :: proc(
+		target_task:    u32,
+		address:        ^u64,
+		size:           u64,
+		mask:           u64,
+		flags:          i32,
+		object:         i32,
+		offset:         uintptr,
+		copy:           b32,
+		cur_protection,
+		max_protection: i32,
+		inheritance:    u32,
+	) -> i32 ---
+}
+
+// The following features are specific to Darwin only.
+VM_FLAGS_ANYWHERE :: 0x0001
+
+SUPERPAGE_SIZE_ANY :: 1
+SUPERPAGE_SIZE_2MB :: 2
+
+VM_FLAGS_SUPERPAGE_SHIFT    :: 16
+VM_FLAGS_SUPERPAGE_SIZE_ANY :: SUPERPAGE_SIZE_ANY << VM_FLAGS_SUPERPAGE_SHIFT
+VM_FLAGS_SUPERPAGE_SIZE_2MB :: SUPERPAGE_SIZE_2MB << VM_FLAGS_SUPERPAGE_SHIFT
+
+MEMORY_OBJECT_NULL :: 0
+VM_PROT_READ  :: 0x01
+VM_PROT_WRITE :: 0x02
+VM_INHERIT_SHARE :: 0
```

Comment:
I see you use `MAP_ANONYMOUS` on other targets, but use `VM_INHERIT_SHARE` here. The equivalent to `MAP_ANONYMOUS` would be `VM_INHERIT_COPY` afaict.

Reply:
This could be why the Intel CI is failing once threading is involved but I can't confirm because I don't have an Intel machine.

Reply:
As silly as this sounds, why not just have different flags for Intel and Arm?